From e9bb095205d71c5ceedbcbb5fbfb7649967b2578 Mon Sep 17 00:00:00 2001
Message-ID: <e9bb095205d71c5ceedbcbb5fbfb7649967b2578.1767632279.git.lorenzo@kernel.org>
In-Reply-To: <317926bb5f936e2aa70ae6dd51c0e5eb918ae3ad.1767632279.git.lorenzo@kernel.org>
References: <317926bb5f936e2aa70ae6dd51c0e5eb918ae3ad.1767632279.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Sat, 27 Dec 2025 09:27:21 +0100
Subject: [PATCH net-next 2/2] net: airoha: Always use qdma2 for GDM{2,3,4}
 ports

Always use qdma2 for GDM2, GDM3 and GDM4 in order to improve traffic load-balance.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 drivers/net/ethernet/airoha/airoha_eth.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -3005,7 +3005,7 @@ static int airoha_setup_phylink(struct n
 #endif
 
 static int airoha_alloc_gdm_port(struct airoha_eth *eth,
-				 struct device_node *np, int index)
+				 struct device_node *np)
 {
 	const __be32 *id_ptr = of_get_property(np, "reg", NULL);
 	struct airoha_gdm_port *port;
@@ -3040,7 +3040,7 @@ static int airoha_alloc_gdm_port(struct
 		return -ENOMEM;
 	}
 
-	qdma = &eth->qdma[index % AIROHA_MAX_NUM_QDMA];
+	qdma = &eth->qdma[!!p];
 	dev->netdev_ops = &airoha_netdev_ops;
 	dev->ethtool_ops = &airoha_ethtool_ops;
 	dev->max_mtu = AIROHA_MAX_MTU;
@@ -3184,7 +3184,6 @@ static int airoha_probe(struct platform_
 	for (i = 0; i < ARRAY_SIZE(eth->qdma); i++)
 		airoha_qdma_start_napi(&eth->qdma[i]);
 
-	i = 0;
 	for_each_child_of_node(pdev->dev.of_node, np) {
 		if (!of_device_is_compatible(np, "airoha,eth-mac"))
 			continue;
@@ -3192,7 +3191,7 @@ static int airoha_probe(struct platform_
 		if (!of_device_is_available(np))
 			continue;
 
-		err = airoha_alloc_gdm_port(eth, np, i++);
+		err = airoha_alloc_gdm_port(eth, np);
 		if (err) {
 			of_node_put(np);
 			goto error_napi_stop;
